src/rules/no-extraneous-dependencies.ts

```diff
@@ -328,7 +328,7 @@ function reportIfMissing(
   })
 }
 
-function testConfig(config: string[] | boolean | undefined, filename: string) {
+function testConfig(config: string[] | boolean | undefined, filename: string, context: RuleContext) {
```

Comment:
```suggestion
function testConfig(config: string[] | boolean | undefined, context: RuleContext) {
```

Read `context.physicalFilename` inside.